### PR TITLE
[User's Guide] Update discussion of C++ code gen

### LINKED ISSUE #276: Add documentation of new C++ code gen to User's Guide
- [ ] Document C++ component base class code gen
- [ ] Document C++ implementation template code gen

### PR DESCRIPTION
**Pull request by bocchino**
Closes #276.

### CODE REVIEW

File: docs/users-guide/Analyzing-and-Translating-Models.adoc [hunk 1]
```diff
@@ -562,12 +540,24 @@ translating constant definitions>>.
...
+Prime User's Guide].
+
+==== Component Implementation Templates
 
-We are phasing in direct {cpp} code generation for types, ports, and
-components.
-Currently we have implemented direct {cpp} code generation for types
-(arrays, structs, and enums) and ports.
+`fpp_to_cpp` has an option `-t`.
+When you run `fpp_to_cpp` with this option, it does not
```
Comment (tiffany1618) [inline docs/users-guide/Analyzing-and-Translating-Models.adoc:554]: I think this should be `fpp-to-cpp` instead of  `fpp_to_cpp`.

**Review by tiffany1618 — COMMENTED** [review]
Looks good! Just one small comment.

File: docs/users-guide/Analyzing-and-Translating-Models.adoc (hunk 1 above)
Comment (bocchino) [reply docs/users-guide/Analyzing-and-Translating-Models.adoc]: Good catch, fixed!

**Review by tiffany1618 — APPROVED** [review]
Looks good!

**Pull request merged**